### PR TITLE
Onnx models

### LINKED ISSUE #34: `ruff` linting
Docs: https://github.com/charliermarsh/ruff

### LINKED ISSUE #42: Remove `tensorflow` dependency
- Port models to `onnx`
- Validate quality
- Validate performance

### PR DESCRIPTION
**Pull request by valentynbez**
- Models ported to ONNX
- Comparison notebook and convert code attached to the project
- `Predictior` rewritten as a part of the package, DeepFRI submodule removed
- tensorflow & biopython dependencies removed (closes #42)
- Automatic switches between GPU and CPU, multithreading on CPU
- new CLI 
- project structure improvements
- new formatting (closes #34)

### CODE REVIEW

**Comment by valentynbez** [comment]
This is the final commit on this PR. This is a working version with onnx models.
Ready to merge from my side.

File: mDeepFRI/CPP_lib/load_contact_maps.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+#include <iostream>
+#include <queue>
+#include <string>
+#include <sys/stat.h>
+
+#include "atoms_file_io.h"
+#include "load_contact_maps.h"
+#include "python_utils.h"
+
+<<<<<<< HEAD
```
Comment (SoliareofAstora) [inline mDeepFRI/CPP_lib/load_contact_maps.cpp:11]: unmerged conflict

File: mDeepFRI/__init__.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+__version__ = "2.0.0"
+model_links = [
+    "https://huggingface.co/crusher083/mDeepFRI/resolve/main/DeepCNN-MERGED_bp.onnx",
```
Comment (SoliareofAstora) [inline mDeepFRI/__init__.py:3]: How about move repeating pattern of huggingface url to some constant?

File: mDeepFRI/cli.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+@click.group()
+@click.option("--debug/--no-debug", default=False)
+@click.version_option(version=__version__)
+@click.pass_context
+def main(ctx, debug):
```
Comment (SoliareofAstora) [inline mDeepFRI/cli.py:19]: I'm more used to main function definition to be placed at the bottom of the file

File: mDeepFRI/predict.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    'Protein': prot,
+                    'GO_term/EC_number': row[0],
+                    'Score': '{:.5f}'.format(row[2]),
+                    'GO_term/EC_number name': row[1]
+                })
+
+        with open(output_fn, 'w', encoding="utf-8") as f:
+            json.dump(data, f, indent=4)
+
+    def save_file(self, output_fn: str, delimiter: str, quotechar: str = '"'):
```
Comment (SoliareofAstora) [inline mDeepFRI/predict.py:117]: I propose make it private

File: weight_convert/convert_models2onnx.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                               opset=15)
+
+
+# Newest CPU models
+
+if __name__ == "__main__":
+
+    args = parse_args()
+
+    model_path = args.weights
```
Comment (SoliareofAstora) [inline weight_convert/convert_models2onnx.py:54]: please define main() function

**Review by SoliareofAstora — CHANGES_REQUESTED** [review]
Piękne zmiany. Nie mogę się doczekać merge

**Review by SoliareofAstora — APPROVED** [review]

**Pull request merged**